igor-monitor-travis/src/main/java/com/netflix/spinnaker/igor/travis/service/TravisService.java

```diff
@@ -444,27 +444,21 @@ private Optional<String> getAndCacheJobLog(int jobId) {
         travisCache.setJobLog(groupKey, jobId, v3Log.getContent());
         return Optional.of(v3Log.getContent());
       }
-    } catch (RetrofitError e) {
-      if (e.getBody() != null) {
-        try {
-          Map body = (Map) e.getBodyAs(HashMap.class);
-          if ("log_expired".equals(body.get("error_type"))) {
-            log.info(
-                "{}: The log for job id {} has expired and the corresponding build was ignored",
-                groupKey,
-                jobId);
-          } else {
-            log.warn(
-                "{}: Could not get log for job id {}. Error from Travis:\n{}",
-                groupKey,
-                jobId,
-                body);
-          }
-        } catch (RuntimeException ex) {
-          log.warn("{}: Could not parse original error message from Travis", groupKey, ex);
+    } catch (SpinnakerServerException e) {
+      if (e instanceof SpinnakerHttpException) { // only SpinnakerHttpException has a response body
```

Comment:
What you've got here matches the current behavior, but...I'm tempted to clean up this code and change the behavior all at the same time.

If I'm reading the code correctly, the current behavior swallows an RetrofitError that has a null body.  That seems a little unfortunate.

If we change this to catch SpinnakerHttpException instead of SpinnakerServerException, we let the other exception types bubble up.  That feels like better behavior.  But, this PR has lived a long time, and I don't have a good way to exercise this specific functionality so let's leave this for another time.